### PR TITLE
refactor: [M3-6669] - Replaced deprecated keycodes

### PR DESCRIPTION
**Pull request by tyler-akamai**
## Description 📝
Replaced deprecated keycodes.

## Major Changes 🔄
event.keyCode === 13
becomes
event.key === 'Enter'

Link for conversions: https://developer.mozilla.org/en-US/docs/Web/API/UI_Events/Keyboard_event_key_values

## How to test 🧪
1. SearchBar.tsx: In the search bar at the top of the page, type the name of one of your Linodes, hover over it, and then hit the `Enter` key. You should navigate to the details page.
2. ActionMenu.tsx: Navigate to the `Linodes` page, hit the Action Menu (`...`) for any linode, hover over any option, and hit the `Enter` key. That option should be executed.
3. ProcessesTable.tsx: Navigate to the `Longview` page. Click `View Details` on any of the Longviews. Navigate to the `Processes` tab. Hover over any process and try to hit the 'Enter' key. This functionality wasn't working pre-changes and still is not working. Refer to comment down below for more information.

### CODE REVIEW

File: packages/manager/src/features/TopMenu/SearchBar/SearchBar.tsx [hunk 1]
```diff
@@ -244,7 +244,7 @@ export const SearchBar = (props: CombinedProps) => {
 
   const onKeyDown = (e: any) => {
     if (
-      e.keyCode === 13 &&
+      e.key === 'Enter' &&
```
Comment (tyler-akamai) [inline packages/manager/src/features/TopMenu/SearchBar/SearchBar.tsx:247]: So this functionality was not working and is still not working.

Things I have tried:
1. Adding `tabIndex={0}` to make the row focusable - only after selecting a single row, you can hover over and press `Enter`. The onKeyUp function will get executed; however, the page does not re-render.
2. `onKeyUp = {() => setSelectedProcess({ name, user })}` - This is identical to the onClick which is working, but still no luck.
3. Replaced `setSelectedProcess={setSelectedProcess}` in `ProcessLanding.tsx` with `setSelectedProcess={() => console.log('Here')}` to ensure the function is getting called. It is. So I believe there is a re-rendering issue that doesn't update and highlight the row after you hit the 'Enter' key.

**Review by abailly-akamai — APPROVED** [review]

File: packages/manager/src/features/TopMenu/SearchBar/SearchBar.tsx (hunk 1 above)
Comment (abailly-akamai) [reply packages/manager/src/features/TopMenu/SearchBar/SearchBar.tsx:247]: @tyler-akamai are you testing the proper behavior? This is when you type something and press enter without using any search suggestion, which should take you to the search result page.

If that's what you are describing as broken, can you please add your machine, OS and browser details? I am not able to reproduce an issue here

**Review by bnussman-akamai — APPROVED** [review]

File: packages/manager/src/features/Longview/LongviewDetail/DetailTabs/Processes/ProcessesTable.tsx [hunk 2]
```diff
@@ -218,7 +218,7 @@ export const ProcessesTableRow: React.FC<ProcessTableRowProps> = React.memo(
     return (
       <TableRow
         onKeyUp={(e: any) =>
-          e.keyCode === 13 && setSelectedProcess({ name, user })
+          e.key === 'Enter' && setSelectedProcess({ name, user })
```
Comment (tyler-akamai) [inline packages/manager/src/features/Longview/LongviewDetail/DetailTabs/Processes/ProcessesTable.tsx:221]: So this functionality was not working and is still not working.

Things I have tried:
1. Adding tabIndex={0} to make the row focusable - only after selecting a single row, you can hover over and press Enter and onKeyDown will execute. However, the row you hit Enter on does not get highlighted.
2. Replaced the onKeyUp function to onKeyUp = {() => setSelectedProcess({ name, user })} - This is identical to the onClick which is working, but still no luck.
3. Replaced setSelectedProcess={setSelectedProcess} in ProcessLanding.tsx with setSelectedProcess={() => console.log('Here')} to ensure the function is getting called. It is. So I believe there is a re-rendering issue that doesn't update and highlight the row after you hit the 'Enter' key.

I am still looking into a solution but feel free to add feedback if you might know the problem.

Comment (tyler-akamai) [reply packages/manager/src/features/Longview/LongviewDetail/DetailTabs/Processes/ProcessesTable.tsx:221]: It was suggested to add a separate ticket referencing this issue.

**Pull request merged**